modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,144 @@
+
+[NOTE]
+====
+If neither of the last two prerequisites in this list are met, establish, or ask someone to establish, the necessary role-based access control (RBAC) so that you can discover `SharedSecret` CR instances and enable service accounts to use `SharedSecret` CR instances.
+====
+
+.Procedure
+
+. Create a `SharedSecret` object instance with the cluster's entitlement secret by running the `oc apply` command:
++
+[IMPORTANT]
+====
+You must have cluster administrator permissions to create `SharedSecret` objects.
+====
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: share-rhel-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a role to grant permission to access the `SharedSecret` object and a `RoleBinding` object that grants permission to access the `SharedSecret` object by running the following command:
++
+[source,terminal]
+----
+$ oc new-project buildah-entitlement
+$ oc apply -f -<<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: use-shared-secret
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - share-entitlement-secret
+    verbs:
+      - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-shared-secret
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: use-shared-secret
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+    namespace: buildah-entitlement
+EOF
+----
+
+. Create a build with relevant mount details using `buildah` buildStrategy:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-rhel-entitlement-build
+spec:
+  source:
+    type: Git
+    git:
+      url: <location of operator's organization>
+      revision: quick-tests
+    contextDir: rh-builds
+  strategy:
+    name: buildah-csi
+    kind: ClusterBuildStrategy
+  paramValues:
+  - name: dockerfile
+    value: Dockerfile
+  volumes:
+  - csi:
+      driver: csi.sharedresource.openshift.io
+      readOnly: true
+      volumeAttributes:
+        sharedSecret: share-entitlement-secret
+    name: rhel-entitlement
+  - name: buildah-auth
+    secret:
+      secretName: buildah-auth
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/buildah-entitlement/sample-go-app
+EOF
+----
+
+. Create the buildRun referencing the rhel-entitlement-build:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: BuildRun
+metadata:
+  name: buildah-rhel-entitlement-buildrun
+spec:
+  build:
+    name: buildah-rhel-entitlement-build
+EOF
+----
```

Comment:
@siamaksade should we instruct developers to use the `shp` command line instead?

```suggestion
. Run a build for `buildah-rhel-entitlement-build`:
+
[source,terminal]
----
$ shp build run buildah-rhel-entitlement-build
----
```

Reply:
Adding about the `shp` command as note